data/ha/hawk_test_driver.py

```diff
@@ -0,0 +1,618 @@
+HREF_NODES = '//a[contains(@href, "#nodes")]'
+HREF_TAGS = '//a[contains(@href, "#tags")]'
+HREF_ALERTS = '//a[contains(@href, "#alerts")]'
+HREF_FENCING = '//a[contains(@href, "#fencing")]'
+RSC_ROWS = "//*[@id=\"resources\"]/div[1]/div[2]/div[2]/table/tbody/tr"
+DROP_DOWN_FORMAT = "//*[@id=\"resources\"]/div[1]/div[2]/div[2]/table/tbody/tr[%d]/td[6]/div/div"
+STONITH_MAINT_ON = '//a[contains(@href, "stonith-sbd/maintenance_on")]'
+NODE_DETAILS = "//*[@id=\"nodes\"]/div[1]/div[2]/div[2]/table/tbody/tr[1]/td[5]/div/a[2]"
+DISMISS_MODAL = "//*[@id=\"modal\"]/div/div/div[3]/button"
+CLEAR_STATE = "//*[@id=\"nodes\"]/div[1]/div[2]/div[2]/table/tbody/tr[1]/td[5]/div/div/button"
+NODE_MAINT = '//a[contains(@href, "maintenance") and contains(@title, "Switch to maintenance")]'
+NODE_READY = '//a[contains(@href, "ready") and contains(@title, "Switch to ready")]'
+GENERATE_REPORT = "//*[@id=\"generate\"]/form/div/div[2]/button"
+CLONE_CHILD = ('//select[contains(@data-help-filter, ".row.resource") and '
+               'contains(@name, "clone[child]")]')
+TARGET_ROLE_FORMAT = ('//select[contains(@class, "form-control") and '
+                      'contains(@name, "%s[meta][target-role]")]')
+STONITH_MAINT_OFF = ('//a[contains(@href, "stonith-sbd") and '
+                     'contains(@title, "Disable Maintenance Mode")]')
+
+class hawkTestDriverError(Exception):
+    """Base class for exceptions in this module."""
+    def __init__(self, value):
+        self.value = value
+
+    def __str__(self):
+        return repr(self.value)
+
+class hawkTestDriver:
+    def __init__(self, addr='localhost', port='7630', browser='firefox', version='12-SP2'):
+        self.set_addr(addr)
+        self.set_port(port)
+        self.timeout_scale = 1
+        self.set_browser(browser)
+        self.driver = ''
+        self.retval = 0
+        self.test_version = str(version)
+
+    def set_addr(self, addr):
+        if isinstance(addr, str):
+            self.addr = addr
+        else:
+            raise hawkTestDriverError('Unexpected type for host address')
+
+    def set_port(self, port):
+        port = str(port)
+        if port.isdigit() and 1 <= int(port) <= 65536:
+            self.port = port
+        else:
+            raise hawkTestDriverError('Port must be an integer')
+
+    def set_browser(self, browser):
+        browser = browser.lower()
+        if browser in ['chrome', 'chromium', 'firefox']:
+            self.browser = browser
+            if browser == 'firefox':
+                self.timeout_scale = 2.5
+            else:
+                self.timeout_scale = 1
+        else:
+            raise hawkTestDriverError('Browser must be chrome, chromium or firefox')
+
+    def set_retval(self, value):
+        self.retval = int(value)
+
+    def get_retval(self):
+        return self.retval
+
+    def _connect(self):
+        if self.browser in ['chrome', 'chromium']:
+            self.driver = webdriver.Chrome()
```

Comment:
as far as I know there is ability to execute chromium , but you just using same driver in both cases , so why you need two options ? just simplify to single option to not confuse people and have less code :)

Reply:
Well, as I said in the other PR, we initially got this as 2 scripts (one for chrome, the other for Firefox) from QAM. When I noticed the code was mostly the same, I merged both scripts into a single one where the driver/browser can be decided at run time via the '-b' browser parameter. Adding to that, there were some tests that due to caching issues (certainly not the case anymore) only worked in Chrome, so the initial idea almost a year ago was to have this run the reduced test scope using Firefox (which was the initial request) and the extended tests using Chrome to try to cover more of the GUI. Nowadays the code is already there, and even though we can do the full test with Firefox, I feel that it's something that could be useful further down the line, or if somebody would like to download this and test by hand.

Reply:
probably you miss my point here . it is normal to have distinction between chrome and firefox in such tests. My question was about why you want to support chrome and chromium I mean you still can run on both but use single word for this

Reply:
ah just find answer on this question below , so you may ignore this comment